core/src/test/scala/akka/kafka/internal/ProducerTest.scala

```diff
@@ -54,14 +46,16 @@ class ProducerTest(_system: ActorSystem)
   implicit val stageStoppingTimeout = StageStoppingTimeout(15.seconds)
   implicit val ec = _system.dispatcher
 
+  val checksum = new java.lang.Long(-1)
+
   type K = String
   type V = String
   type Record = ProducerRecord[K, V]
   type Msg = Message[String, String, NotUsed.type]
 
   def recordAndMetadata(seed: Int) = {
     new ProducerRecord("test", seed.toString, seed.toString) ->
-      new RecordMetadata(new TopicPartition("test", seed), seed.toLong, seed.toLong, Record.NO_TIMESTAMP, -1, -1, -1)
+      new RecordMetadata(new TopicPartition("test", seed), seed.toLong, seed.toLong, RecordBatch.NO_TIMESTAMP, checksum, -1, -1)
```

Comment:
I suggest not relying on the internal `RecordBatch` class. Set your own constant or hardcode it since it's a test.